core/src/services/azfile/core.rs

```diff
@@ -417,8 +417,8 @@ impl AzfileCore {
         // azure file service does not support recursive directory creation
         let mut p = path;
         while p != "/" {
-            p = get_parent(p);
             dirs.push_front(p);
+            p = get_parent(p);
```

Comment:
If input path is `/a/b/c`, will this be wrong?

Reply:
Thanks for review.

The initial intent here is to avoid puting `"/"` into the `dirs`, but ignoring the usage of the function, sry for that.

Since the [document](https://learn.microsoft.com/en-us/rest/api/storageservices/create-directory#request) shows that the `Create Directory` request requires to format the dir like `...path/to/dir...`, the problem may lie in the trailing `/` of the input path. Is it a workable option to just trim the trailing `/`?

Still I'm not sure the meaning of `normalize the root` from the issue since I haven't found something wrong about it. Could you give me a hint?

Sry for taking a lot of your time as a newcomer and thanks for your patience.

Reply:
> Still I'm not sure the meaning of `normalize the root` from the issue since I haven't found something wrong about it. Could you give me a hint?

normalize root is about the code here:

https://github.com/apache/incubator-opendal/blob/0711e6ce97660380628b72a010d092fd29ec9553/core/src/services/azfile/backend.rs#L161

But we have already done that. So it might not be the root cause.

Reply:
Perhaps we should directly test the azfile service for the error message.